java/arcs/core/data/Capabilities.kt

```diff
@@ -65,6 +61,14 @@ class Capabilities(capabilities: List<Capability> = emptyList()) {
         return other.ranges.all { otherRange -> contains(otherRange) }
     }
 
+    fun isEquivalent(other: Capabilities): Boolean {
+        return ranges.size == other.ranges.size && other.ranges.all { hasEquivalent(it) }
+    }
+
+    fun hasEquivalent(capability: Capability): Boolean {
+        return ranges.find { it.isCompatible(capability) && it.isEquivalent(capability) } != null
```

Comment:
Possibly you want `ranges.any { ... }` instead?

Reply:
right, thanks! done!